### PR TITLE
EKF: Print reason for EKF2 arming failure when no GPS is available.

### PR DESCRIPTION
**Pull request by samuelctabor**
Previously this situation returned an empty string for the EKF failure reason, so the message was "APM: PreArm: ". Now it is "APM: PreArm: EKF2 initialisation failure: No GPS lock".
I also updated the description of the AHRS_GPS_USE parameter, as it appears to only affect the DCM AHRS.

### CODE REVIEW

**Comment by rmackay9** [comment]
OK, i'd never seen this.  I've tagged @priseborough so hopefully he can give the "OK" and we can merge.  thanks!

File: libraries/AP_AHRS/AP_AHRS.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ const AP_Param::GroupInfo AP_AHRS::var_info[] = {
 
     // @Param: GPS_USE
     // @DisplayName: AHRS use GPS for navigation
-    // @Description: This controls whether to use dead-reckoning or GPS based navigation. If set to 0 then the GPS won't be used for navigation, and only dead reckoning will be used. A value of zero should never be used for normal flight.
+    // @Description: This controls whether to use dead-reckoning or GPS based navigation. If set to 0 then the GPS won't be used for navigation, and only dead reckoning will be used. A value of zero should never be used for normal flight. Currently this affects only the DCM-based AHRS: GPS is required for the EKF-based AHRS to function.
```
Comment (priseborough) [inline libraries/AP_AHRS/AP_AHRS.cpp]: This is not true. The EKF can function without GPS.

**Comment by priseborough** [comment]
I don't agree with the second patch. The EKF can function without GPS and will produce an attitude, vertical speed and vertical position estimate.

**Comment by samuelctabor** [comment]
Thanks Paul. On closer reading it looks like EKF only requires GPS on ArduPlane, is that correct? Anyway I couldn't find any usage of AHRS_GPS_USE parameter in the EKF AHRS.

**Comment by priseborough** [comment]
It is not an EKF requirement. It is a requirement of the plane arming checks that the EKF is producing  a global position, which the EKF it won't do unless it has started using GPS.

**Comment by rmackay9** [comment]
Merged although I have a couple of questions/concerns:

- will this message only be printed once or will it be printed over and over at a very high rate until initialisation happens?
- shouldn't we have a similar change made for EKF3?

**Comment by samuelctabor** [comment]
- It will only print once, when arming is requested, similar to the other messages.
- For EKF3 the equivalent logic is in AP_NavEKF3_core.cpp:401. I can commit the change if that's easier.

**Comment by rmackay9** [comment]
@samuelctabor, great, thanks for the answers.
If you can do a PR for EKF3 as well, that'd be great.  We're moving to the EKF3 for Copter-3.6.

**Pull request closed without merging**